actions/drain_node_handler.go

```diff
@@ -39,10 +44,29 @@ func newDrainNodeHandler(log logrus.FieldLogger, clientset kubernetes.Interface)
 			podDeleteRetryDelay:           5 * time.Second,
 			podEvictRetryDelay:            5 * time.Second,
 			podsTerminationWaitRetryDelay: 10 * time.Second,
+			castNamespace:                 castNamespace,
+			skipDeletedTimeoutSeconds:     60,
 		},
 	}
 }
 
+// getDrainTimeout returns drain timeout adjusted to action creation time.
+// After 2 minutes grace period action timeout is adjusted using the time passed between action creation and now
+// the result is clamped between 60s and the requested timeout.
+func (h *drainNodeHandler) getDrainTimeout(action *castai.ClusterAction) time.Duration {
+	timeSinceCreated := time.Since(action.CreatedAt)
+	drainTimeout := time.Duration(action.ActionDrainNode.DrainTimeoutSeconds) * time.Second
+	// Allow to have 2 minutes grace period for newer actions.
+	if timeSinceCreated < 2*time.Minute {
+		return drainTimeout
+	}
+
+	// Remove time required for polling the action.
+	timeSinceCreated = timeSinceCreated - 100*time.Second
```

Comment:
100 Seconds is quite a lot?
Also we have actions priority which might impact when action is picked up during times when there's a long queue of actions.